ios/brave-ios/Sources/Brave/Helpers/URLBarHelper.swift

```diff
@@ -0,0 +1,1322 @@
+// Copyright 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import Foundation
+import UIKit
+
+class URLBarHelper {
+
+  static let shared = URLBarHelper()
+
+  func shouldShowSearchSuggestions(using lastReplacement: String) -> Bool {
+    // Check if last entry to url textfield needs to be checked as suspicious.
+    // The reason of checking count is bigger than 1 is the single character
+    // entries will always be safe and only way to achieve multi character entry is
+    // using paste board.
+    // This check also allow us to handle paste permission case
+    guard lastReplacement.count > 1 else {
+      return true
+    }
+
+    // Check if paste board has any text to guarantee the case
+    guard UIPasteboard.general.hasStrings || UIPasteboard.general.hasURLs else {
+      return true
+    }
+
+    // Perform check on pasted text
+    if let pasteboardContents = UIPasteboard.general.string {
```

Comment:
I tested and this does not trigger the OS prompt, is it because the user pastes the string explicitly beforehand?

Reply:
Any UIPasteBoard interaction should trigger the permission from Os if user didnt allow this permission that particular app before.

And for our case it doesnt matter cause first guard statement is actually checking multi character entry case.